devito/types/grid.py

```diff
@@ -873,7 +873,7 @@ def __subdomain_finalize_core__(self, grid):
         # SubDomainSet ID to make them unique so they can be used to key a dictionary
         # of replacements without risking overwriting.
         i_dim = Dimension(f'n_{str(id(self))}')
-        d_dim = DefaultDimension(name='d', default_value=2*grid.dim)
+        d_dim = DefaultDimension(name='sds_dim', default_value=2*grid.dim)
```

Comment:
bit nitpicky: I think ideally, grid should have a `grid_registry` at the top for this, and similarly probably a `sparse_registry` in sparse.

Reply:
I don't like the idea of having any non-compiler-scoped symbol registry. Registries/counters attached to classes make a huge mess when you have test sets or multiple operators within a python runtime and you end up with massive increments in all the names, plus it can make test sets order-dependant if you're not careful.

Reply:
>Registries/counters attached to classes

No not attached to a class. It's just a global registry for the file that all class instantiated from the class definition in that file uses. That's how the registry is used everywhere.